File: vigil-backend/app/schemas/pull_request.py
from datetime import datetime
from typing import List, Optional
import uuid
from pydantic import BaseModel, ConfigDict
from app.schemas.base import PaginatedResponse


class PullRequestRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    repository_id: uuid.UUID
    github_pr_id: int
    pr_number: int
    title: str
    description: Optional[str] = None
    author_login: str
    source_branch: str
    target_branch: str
    head_sha: str
    base_sha: str
    status: str
    created_at: datetime
    updated_at: datetime
    closed_at: Optional[datetime] = None
    merged_at: Optional[datetime] = None


class PullRequestListResponse(PaginatedResponse):
    items: List[PullRequestRead]

File: vigil-frontend/src/types/pullRequest.ts
import type { PaginatedResponse } from './api';

export interface PullRequestRead {
  id: string; // uuid
  repository_id: string; // uuid
  github_pr_id: number;
  pr_number: number;
  title: string;
  description: string | null;
  author_login: string;
  source_branch: string;
  target_branch: string;
  head_sha: string;
  base_sha: string;
  status: string; // e.g., "OPEN", "CLOSED", "MERGED"
  created_at: string;
  updated_at: string;
  closed_at: string | null;
  merged_at: string | null;
}

export type PullRequestListResponse = PaginatedResponse<PullRequestRead>;
